### PR TITLE
More swap wip

### PR DESCRIPTION
**Pull request by bubafistah**
Address parsing, tx constructing wip, etc.
Based on [https://github.com/cryptozoidberg/boolberry](url)

### CODE REVIEW

File: src/cryptonote_core/cryptonote_tx_utils.cpp [hunk 1]
```diff
@@ -289,10 +295,20 @@ namespace cryptonote
...
       tx_out out;
       out.amount = dst_entr.amount;
-      txout_to_key tk;
-      tk.key = out_eph_public_key;
+
+      txout_to_key tk = AUTO_VAL_INIT(tk);
+
+      if (dst_entr.addr.is_swap_addr) {
+        // Zero key for swap txs
+        // TODO Should probably do all address math only when neccessary
```
Comment (Snider) [inline src/cryptonote_core/cryptonote_tx_utils.cpp:303]: Could we do an impossible key, like `000000000000000000000000000000000000000` ?

File: tests/unit_tests/swap_test.cpp [hunk 2]
```diff
@@ -17,6 +18,46 @@ TEST(swap_address, swap_addr_extra_userdata_entry_from_addr)
   ASSERT_TRUE(entry.is_checksum_valid());
 }
 
+TEST(swap_address, swap_parse_addr_from_str)
```
Comment (Snider) [inline tests/unit_tests/swap_test.cpp:21]: Yay, code with a test, we are like adults now... my word, what happened :P

**Review by Snider — APPROVED** [review]
fank oooo <3

**Comment by Snider** [comment]
are you able to merge @bubafistah ? if not its time you get super cow powers.

File: src/cryptonote_core/cryptonote_tx_utils.cpp (hunk 1 above)
Comment (bubafistah) [reply src/cryptonote_core/cryptonote_tx_utils.cpp:303]: Not sure, for now will just implement boolberry logic. Will have some free time by the end of the week, then i will give some more precise answer.

**Comment by Snider** [comment]
@bubafistah, well, I'm going to get out of your way; time for you to spread your wings and be the blockchain lead ( I'm still here to make you'll smile and review changes)

Need me to do anything, ask, and it won't be a problem.

**Pull request merged**